lib/kafkaClient.js

```diff
@@ -241,6 +241,37 @@ KafkaClient.prototype.connectToBroker = function (broker, callback) {
   }, timeout);
 };
 
+KafkaClient.prototype.getController = function (callback) {
+  this.loadMetadataForTopicsV2([], (error, result) => {
+    if (error) {
+      return callback(error);
+    }
+
+    var controllerId = result[1].clusterMetadata.controllerId;
```

Comment:
how often does the `controllerId` change?  Would it be safe to cache this on the client so we can avoid doing a metadata request?

Reply:
I have not been able to find documentation that verifies this - my impression is that the election process happens when starting the cluster and otherwise if the current elected broker crashes/becomes unavailable. We can cache it, and update metadata if the request ends up being sent to a broker that isn't controller?